packages/edit-post/src/components/visual-editor/test/index.native.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import { initializeEditor, fireEvent } from 'test/helpers';
+
+/**
+ * WordPress dependencies
+ */
+import { getBlockTypes, unregisterBlockType } from '@wordpress/blocks';
+import { registerCoreBlocks } from '@wordpress/block-library';
+
+beforeAll( () => {
+	// Register all core blocks
+	registerCoreBlocks();
+} );
+
+afterAll( () => {
+	// Clean up registered blocks
+	getBlockTypes().forEach( ( block ) => {
+		unregisterBlockType( block.name );
+	} );
+} );
+
+const initialHtml = `
+<!-- wp:paragraph -->
+<p>First example paragraph.</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p>Second example paragraph.</p>
+<!-- /wp:paragraph -->
+`;
+
+describe( 'when title is focused', () => {
+	it( 'new blocks are inserted after the title', async () => {
+		const {
+			getByA11yLabel,
+			getAllByA11yLabel,
+			getByText,
+		} = await initializeEditor( {
+			initialHtml,
+		} );
+
+		// Focus first block
+		fireEvent.press( getByA11yLabel( /Paragraph Block. Row 1/ ) );
+
+		// Focus title
+		fireEvent( getAllByA11yLabel( 'Post title. test' )[ 0 ], 'select' );
+
+		// Add new Heading block
+		fireEvent.press( getByA11yLabel( 'Add block' ) );
```

Comment:
I think you might need to include the following code to make the `FlatList` of the inserter menu's blocklist render the items:
```
		const blockList = getByTestId( 'InserterUI-Blocks' );
		// onScroll event used to force the FlatList to render all items
		fireEvent.scroll( blockList, {
			nativeEvent: {
				contentOffset: { y: 0, x: 0 },
				contentSize: { width: 100, height: 100 },
				layoutMeasurement: { width: 100, height: 100 },
			},
		} );
```

Reply:
Yeah, it is working currently as the Heading block is generally one of the first block types displayed and therefore scrolling is unneeded. However, it is possible that one day that may not be the case.

Reply:
Oh that's right, the `FlatList` renders the first items without triggering the scroll event. In that case, we can leave it this way 👍 .